graphql-compiler-plugin/src/main/java/io/ballerina/stdlib/graphql/compiler/GraphqlFunctionValidator.java

```diff
@@ -113,6 +130,19 @@ private void validateReturnType(TypeSymbol returnTypeDesc,
             if (typeReferenceTypeSymbol.definition().kind() == SymbolKind.TYPE_DEFINITION) {
                 TypeDefinitionSymbol typeDefinitionSymbol =
                         (TypeDefinitionSymbol) typeReferenceTypeSymbol.definition();
+                if (typeReferenceTypeSymbol.typeDescriptor().typeKind() == TypeDescKind.RECORD) {
+                    RecordTypeSymbol recordTypeSymbol = (RecordTypeSymbol) typeReferenceTypeSymbol.typeDescriptor();
+                    Map<String, RecordFieldSymbol>  recordFieldSymbolMap =  recordTypeSymbol.fieldDescriptors();
+                    for (Map.Entry<String, RecordFieldSymbol> entry:recordFieldSymbolMap.entrySet()) {
+                        if (entry.getValue().typeDescriptor().typeKind().equals(TypeDescKind.TYPE_REFERENCE)) {
+                            validateReturnType(entry.getValue().typeDescriptor(), location, context);
+                        } else {
+                            if (PluginUtils.isInvalidPath(entry.getKey())) {
```

Comment:
This seems to be a generic check, not something specific to resource paths? If that's the case, shall we rename this method? It's a bit misleading